### ISSUE TITLE
Actions CI: Unittests build fails for g++ (Ubuntu 7.5.0-3ubuntu1~18.04) 7.5.0

### ISSUE THREAD

**Issue by Shreeshrii** [tesseract-ocr/tesseract#3360]
Please see 

https://github.com/tesseract-ocr/tesseract/runs/2169172785?check_suite_focus=true#step:10:7

https://github.com/Shreeshrii/tesseract/runs/2171938618?check_suite_focus=true#step:6:45

https://github.com/Shreeshrii/tesseract/runs/2171938600?check_suite_focus=true#step:6:45

**Comment by Shreeshrii** [comment]
```
  CXX      src/ccmain/libtesseract_la-par_control.lo
src/ccmain/par_control.cpp: In member function ‘void tesseract::Tesseract::PrerecAllWordsPar(const std::vector<tesseract::WordData>&)’:
src/ccmain/par_control.cpp:57:21: error: found ‘:’ in nested-name-specifier, expected ‘::’
     for (auto &blob : blobs) {
                     ^
src/ccmain/par_control.cpp:57:16: error: ‘blob’ has not been declared
     for (auto &blob : blobs) {
                ^~~~
src/ccmain/par_control.cpp:57:28: error: qualified-id in declaration before ‘)’ token
     for (auto &blob : blobs) {
                            ^
src/ccmain/par_control.cpp:57:28: error: expected ‘=’ before ‘)’ token
src/ccmain/par_control.cpp:60:3: error: expected ‘;’ before ‘}’ token
   } else {
   ^
src/ccmain/par_control.cpp:60:3: error: expected primary-expression before ‘}’ token
src/ccmain/par_control.cpp:60:3: error: expected ‘;’ before ‘}’ token
```

https://github.com/Shreeshrii/tesseract/runs/2173977915?check_suite_focus=true#step:6:50

**Comment by Shreeshrii** [comment]
https://github.com/Shreeshrii/tesseract/runs/2173977898?check_suite_focus=true#step:6:49
```
  CXX      src/ccmain/libtesseract_la-par_control.lo
src/ccmain/par_control.cpp:57:5: error: statement after '#pragma omp parallel for' must be a for loop
    for (auto &blob : blobs) {
    ^
1 error generated.
Makefile:5372: recipe for target 'src/ccmain/libtesseract_la-par_control.lo' failed
make[2]: Leaving directory '/home/runner/work/tesseract/tesseract'
make[2]: *** [src/ccmain/libtesseract_la-par_control.lo] Error 1
```

**Comment by Shreeshrii** [comment]
I tried again after adding '--disable-openmp' to configure and the build is progressing.
See 
https://github.com/Shreeshrii/tesseract/runs/2174692343?check_suite_focus=true

**Comment by Shreeshrii** [comment]
Problem seems to be in the following code:

```
  // Pre-classify all the blobs.
  if (tessedit_parallelize > 1) {
#ifdef _OPENMP
#  pragma omp parallel for num_threads(10)
#endif // _OPENMP
    for (auto &blob : blobs) {
      *blob.choices = blob.tesseract->classify_blob(blob.blob, "par", ScrollView::WHITE, nullptr);
    }
  } else {
    // TODO(AMD) parallelize this.
    for (auto &blob : blobs) {
      *blob.choices = blob.tesseract->classify_blob(blob.blob, "par", ScrollView::WHITE, nullptr);
    }
  }
```

**Comment by egorpugin** [comment]
Probably won't fix. Use newer gcc.

**Comment by AlexanderP** [comment]
Debian buster
gcc version 8.3.0 (Debian 8.3.0-6)
Failed build

https://gist.github.com/AlexanderP/275d78985cef9772cf20200b9297f15d

```
In file included from ./src/ccstruct/points.h:22,
                 from ./src/ccstruct/blobs.h:25,
                 from src/ccmain/fixspace.cpp:23:
./src/ccutil/elst.h:439:16: warning: 'worst_word_it.tesseract::ELIST_ITERATOR::prev' may be used uninitialized in this function [-Wmaybe-uninitialized]
     prev->next = new_element;
     ~~~~~~~~~~~^~~~~~~~~~~~~
src/ccmain/fixspace.cpp:623:15: note: 'worst_word_it.tesseract::ELIST_ITERATOR::prev' was declared here
   WERD_RES_IT worst_word_it;
               ^~~~~~~~~~~~~
src/ccmain/osdetect.cpp: In member function 'bool tesseract::OrientationDetector::detect_blob(tesseract::BLOB_CHOICE_LIST*)':
src/ccmain/osdetect.cpp:385:25: warning: comparison of integer expressions of different signedness: 'int' and 'std::vector<int>::size_type' {aka 'long unsigned int'} [-Wsign-compare]
           for (s = 0; s < allowed_scripts_->size(); ++s) {
                       ~~^~~~~~~~~~~~~~~~~~~~~~~~~~
src/ccmain/osdetect.cpp: In member function 'void tesseract::ScriptDetector::detect_blob(tesseract::BLOB_CHOICE_LIST*)':
src/ccmain/osdetect.cpp:481:23: warning: comparison of integer expressions of different signedness: 'int' and 'std::vector<int>::size_type' {aka 'long unsigned int'} [-Wsign-compare]
         for (s = 0; s < allowed_scripts_->size(); ++s) {
                     ~~^~~~~~~~~~~~~~~~~~~~~~~~~~
src/ccmain/osdetect.cpp:486:15: warning: comparison of integer expressions of different signedness: 'int' and 'std::vector<int>::size_type' {aka 'long unsigned int'} [-Wsign-compare]
         if (s == allowed_scripts_->size()) {
             ~~^~~~~~~~~~~~~~~~~~~~~~~~~~~
  CXX      src/ccmain/libtesseract_la-superscript.lo
src/ccmain/par_control.cpp: In member function 'void tesseract::Tesseract::PrerecAllWordsPar(const std::vector<tesseract::WordData>&)':
src/ccmain/par_control.cpp:44:28: warning: comparison of integer expressions of different signedness: 'int' and 'std::vector<tesseract::Tesseract*>::size_type' {aka 'long unsigned int'} [-Wsign-compare]
         Tesseract *sub = s < sub_langs_.size() ? sub_langs_[s] : this;
                          ~~^~~~~~~~~~~~~~~~~~~
src/ccmain/par_control.cpp:57:21: error: found ':' in nested-name-specifier, expected '::'
     for (auto &blob : blobs) {
                     ^
                     ::
src/ccmain/par_control.cpp:57:16: error: 'blob' has not been declared
     for (auto &blob : blobs) {
                ^~~~
src/ccmain/par_control.cpp:57:28: error: qualified-id in declaration before ')' token
     for (auto &blob : blobs) {
                            ^
src/ccmain/par_control.cpp:59:6: error: expected ';' before '}' token
     }
      ^
      ;
   } else {
   ~   
src/ccmain/par_control.cpp:60:3: error: expected primary-expression before '}' token
   } else {
   ^
src/ccmain/par_control.cpp:59:6: error: expected ';' before '}' token
     }
      ^
      ;
   } else {
   ~   
src/ccmain/par_control.cpp:60:3: error: expected primary-expression before '}' token
   } else {
   ^
src/ccmain/par_control.cpp:59:6: error: expected ')' before '}' token
     }
      ^
      )
   } else {
   ~   
src/ccmain/par_control.cpp:57:9: note: to match this '('
     for (auto &blob : blobs) {
         ^
src/ccmain/par_control.cpp:60:3: error: expected primary-expression before '}' token
   } else {
   ^
src/ccmain/par_control.cpp:57:5: error: expected iteration declaration or initialization
     for (auto &blob : blobs) {
     ^~~
make[4]: *** [Makefile:5747: src/ccmain/libtesseract_la-par_control.lo] Error 1
make[4]: *** Waiting for unfinished jobs....
make[4]: Leaving directory '/build/tesseract-5.0.0~git5372-1205f036'
make[3]: *** [Makefile:8634: all-recursive] Error 1
make[3]: Leaving directory '/build/tesseract-5.0.0~git5372-1205f036'
make[2]: *** [Makefile:3148: all] Error 2
make[2]: Leaving directory '/build/tesseract-5.0.0~git5372-1205f036'
make[1]: *** [debian/rules:22: override_dh_auto_build] Error 2
make[1]: Leaving directory '/build/tesseract-5.0.0~git5372-1205f036'
make: *** [debian/rules:19: build] Error 2
```

**Comment by egorpugin** [comment]
What about g++-9?

**Comment by stweil** [comment]
I just had the same issue with clang-10 and fixed it in commit 264dfb36859b3383c1337306d3825a1fcf3edc64.

**Comment by AlexanderP** [comment]
Commit 264dfb3 solved the build problem. Thanks!
